src/routes/taskRoute.js

```diff
@@ -15,18 +56,28 @@ router.post("/task/create", async (req, res) => {
 
 router.get("/task/all", async (req, res) => {
   try {
-    const tasks = await Task.find();
-    if (!tasks) res.status(404).send();
+    /* now we have 2 types of data in redis. one json ( tasks ) and another array( groups ). 
+    to get all task we need to extract only json data */
+
+    let ids = await client.scan(0, { TYPE: "ReJSON-RL" }); // return all json type data keys ( all task ids )
```

Comment:
scan is a very slow operation in redis. we use redis sets instead to store the data. instead of scanning all keys